### DISCUSSION TITLE
Do we have prompt mixing keyworks in Fooocus ? (like [Keanu Reeves:Emma Watson:0.4])

### DISCUSSION THREAD

**Discussion by AndreiLungu64** [lllyasviel/Fooocus#1380] [General]
I am wondering if we can mix/add keywords with different weights inside the promp like in A1111

**Comment by lllyasviel** [comment]
The supported one is Midjourney's Multi Prompt.

Also read the doc here:

https://docs.midjourney.com/docs/multi-prompts

Nevertheless, in Fooocus we directly use multiple line rather than `::` . You use multiple line of promts to use it.

For example

    (Keanu Reeves:0.6)
    (Emma Watson:0.4)

(Note that they are in two lines)

Or you can do more complicated mixing

    A photo of (Keanu Reeves:0.6)
    A painting of (Emma Watson:0.4)

(Note that they are in two lines)

**Reply by Phrixus2023** [reply]
Thank you very much for the fooocus, this doesn't feel as if it's stable when using Multi Prompt, e.g. typing 
(cat:0.6)
(fish:0.4)
 but the graphs that come out show no sign of blending, not sure if this is being used incorrectly or if the feature needs to be iterated at the moment?

**Reply by AndreiLungu64** [reply]
It doesn't work for me either

**Comment by SpcCw** [comment]
To me it seems like it kind of works but it is really unstable and dependent on the model. If the concept is known to it we can expect results otherwise it is unpredictable.

For example prompt `young woman in yellow dress holding an (apple: watermelon: 0.1)`
![image](https://github.com/lllyasviel/Fooocus/assets/35988439/cb6e0103-cf20-4fd4-9a00-b3136eb0b25f)

`young woman in yellow dress holding an (apple: watermelon: 0.7)`
![image](https://github.com/lllyasviel/Fooocus/assets/35988439/6d1c2175-c4c3-4902-8e28-5d5529a0ad8e)
So this sort of works except scheduling seems to be random (i.e. it does not diffuse apple for 0.7 of cycles and then continues as watermelon but does mixing from the start). It works a bit better with high number of cycles.

However if the concept is too weird for the model it just does not blend. For example I tried to blend full moon and skull (to get skies with full moon having mild skull features) and couldn't get it - I either get both separately or skull completely dominates the scene and playing with weights changes the result from just the skull in skies to the skull with some moonlight glow.

I would be very interested to get proper prompt scheduling working (as [described here](https://stable-diffusion-art.com/prompt-guide/#Keyword_blending)) but right now it seems easier to use variation/inpaint over existing image.

**Comment by thiner** [comment]
The mixtrue of prompts seems not working with default model. But blending effect is strong when switched to `Blue pencil` model.
```
A photo of (Keanu Reeves:0.6)
A painting of (Emma Watson:0.4)
```
![image (30)](https://github.com/lllyasviel/Fooocus/assets/1897227/ca396394-8395-49c4-8a74-30866cf56a59)

```
mighty (tiger head:0.6)
brown hair (monkey body:0.4), black ears
walking in street
```
![image (31)](https://github.com/lllyasviel/Fooocus/assets/1897227/38ad1885-c76e-49ae-8ef0-20482b9e6870)

**Comment by poor7** [comment]
Multi Prompt does not work.

For example: 
Seed:1234, Styles: 'Fooocus V2', 'Fooocus Enhance', 'Fooocus Sharp'

```
(apple:0.9)
(watermelon:0.1)
```
![2024-01-08_05-55-13_8220](https://github.com/lllyasviel/Fooocus/assets/154618384/892c74d0-4425-451a-aa46-7a9431913bf8)

```
(apple:0.1)
(watermelon:0.9)
```
![2024-01-08_05-52-23_8220](https://github.com/lllyasviel/Fooocus/assets/154618384/e54118cc-4878-4775-84d6-f3cc23a48a33)

```
(watermelon:0.9)
(apple:0.1)
```
![2024-01-08_05-53-08_8220](https://github.com/lllyasviel/Fooocus/assets/154618384/271d1ca6-9a0a-414a-93c8-608448f1cf7a)

```
(watermelon:0.1)
(apple:0.9)
```
![2024-01-08_05-53-54_8220](https://github.com/lllyasviel/Fooocus/assets/154618384/9f5acb7d-c826-4e0a-9dda-5e7fcdf492d6)

Always draws what is specified on the first line, regardless of weight.

This `young woman in white dress holding an (apple:watermelon:0.1)`
![2024-01-08_06-09-20_8220](https://github.com/lllyasviel/Fooocus/assets/154618384/9d973ec9-55b5-4e38-bb09-91e8f4714545)
`young woman in white dress holding an (apple:watermelon:0.9)`
![2024-01-08_06-08-39_8220](https://github.com/lllyasviel/Fooocus/assets/154618384/0613c8b3-9bca-4645-aa96-86d81fe793ec)
doesn't work either

**Reply by thiner** [reply]
Maybe you should select the `blue pencil` model from model list. Seems `blue pencil` is the only workable model for prompt mixing. I can't tell whether it's by design or a bug of Fooocus. @lllyasviel can you please confirm?

**Reply by thiner** [reply]
OK, as per my test, you need to apply settings to have the prompt mixing effect.
1. Uncheck all preset styles.
2. Select the `blue_pencil` model.
3. Write prompts in multi-line format.

An example:
![image](https://github.com/lllyasviel/Fooocus/assets/1897227/dc145c31-2848-41d5-a8cb-777789414c1c)

**Comment by bblackmoor** [comment]
Has there been any change in this functionality in the past month? I am not even sure what to properly call it. Morphing two things into one thing. A111 calls it "attention shift" (https://github.com/lllyasviel/Fooocus/discussions/2322).

**Comment by bsdesigner** [comment]
It blended really well for me:
A photo of (Keanu Reeves:0.6)
A painting of (Emma Watson:0.4)
![Keanu Reeves   Emma Watson](https://github.com/lllyasviel/Fooocus/assets/36730003/21c56336-94db-4a6c-955c-9e786b3ac229)
![изображение_2024-03-12_140300112](https://github.com/lllyasviel/Fooocus/assets/36730003/d01a7273-ffed-4ff3-aa47-1b7bf687e870)

**Reply by bblackmoor** [reply]
That **is** the kind of blending I am looking for. I don't understand why it would be necessary to duplicate the entire prompt just to have one part blended. That seems an inefficient way to parse tokens. Is this _really_ what we are supposed to do in Fooocus?

```
comic in the style of (Chris Claremont:1.6) of (Jaime Pressly:1.6) (1938:1.4) (golden age superhero:1.2) wearing (baggy light blue coveralls:1.3). She has (folding metal airfoils:1.4) and a (hand-stitched costume:1.4) (made of cloth and leather:1.3) with (visible seams:1.6)
comic in the style of (Chris Claremont:1.6) of (Anna Faris:1.6) (1938:1.4) (golden age superhero:1.2) wearing (baggy light blue coveralls:1.3). She has (folding metal airfoils:1.4) and a (hand-stitched costume:1.4) (made of cloth and leather:1.3) with (visible seams:1.6)
comic in the style of (Jim Lee:1.6) of (Jaime Pressly:1.6) (1938:1.4) (golden age superhero:1.2) wearing (baggy light blue coveralls:1.3). She has (folding metal airfoils:1.4) and a (hand-stitched costume:1.4) (made of cloth and leather:1.3) with (visible seams:1.6)
comic in the style of (Jim Lee:1.6) of (Anna Faris:1.6) (1938:1.4) (golden age superhero:1.2) wearing (baggy light blue coveralls:1.3). She has (folding metal airfoils:1.4) and a (hand-stitched costume:1.4) (made of cloth and leather:1.3) with (visible seams:1.6)
```

**Reply by bblackmoor** [reply]
The same prompt with color highlights to show the alternating prompts to be merged.

The intention is that $\color{red}{\textsf{Chris Claremont}}$ and $\color{orange}{\textsf{Jim Lee}}$ are blended, and $\color{green}{\textsf{Jaime Pressly}}$ and $\color{blue}{\textsf{Anna Faris}}$ are blended.

Note that this is all **one prompt**, on multiple lines.

> comic in the style of ($\color{red}{\textsf{Chris Claremont}}$:1.6) of ($\color{green}{\textsf{Jaime Pressly}}$:1.6) (1938:1.4) (golden age superhero:1.2) wearing (baggy light blue coveralls:1.3). She has (folding metal airfoils:1.4) and a (hand-stitched costume:1.4) (made of cloth and leather:1.3) with (visible seams:1.6)
> comic in the style of ($\color{red}{\textsf{Chris Claremont}}$:1.6) of ($\color{blue}{\textsf{Anna Faris}}$:1.6) (1938:1.4) (golden age superhero:1.2) wearing (baggy light blue coveralls:1.3). She has (folding metal airfoils:1.4) and a (hand-stitched costume:1.4) (made of cloth and leather:1.3) with (visible seams:1.6)
> comic in the style of ($\color{orange}{\textsf{Jim Lee}}$:1.6) of ($\color{green}{\textsf{Jaime Pressly}}$:1.6) (1938:1.4) (golden age superhero:1.2) wearing (baggy light blue coveralls:1.3). She has (folding metal airfoils:1.4) and a (hand-stitched costume:1.4) (made of cloth and leather:1.3) with (visible seams:1.6)
> comic in the style of ($\color{orange}{\textsf{Jim Lee}}$:1.6) of ($\color{blue}{\textsf{Anna Faris}}$:1.6) (1938:1.4) (golden age superhero:1.2) wearing (baggy light blue coveralls:1.3). She has (folding metal airfoils:1.4) and a (hand-stitched costume:1.4) (made of cloth and leather:1.3) with (visible seams:1.6)

**Reply by bsdesigner** [reply]
Try doing it with an array::
![изображение_2024-03-12_212020302](https://github.com/lllyasviel/Fooocus/assets/36730003/7a83b1fb-d728-41f0-a1f4-b344d173f9b5)

**Reply by bblackmoor** [reply]
Thank you. Do you have a link handy for the documentation of how that works?

**Reply by bsdesigner** [reply]
I haven't come across any such documentation. Just experimenting with settings and prompts...

**Comment by bblackmoor** [comment]
It seems weird to me, but it does seem to work. This prompt mixes $\color{red}{\textsf{Anya Taylor-Joy}}$, $\color{purple}{\textsf{Amanda Seyfried}}$, $\color{blue}{\textsf{Cara Delevingne}}$, and $\color{green}{\textsf{Adriana Lima}}$.

> Dramatic illustration (art by Todd McFarlane:1.6) of ($\color{red}{\textsf{Anya Taylor-Joy}}$:1.4) as a programmer (standing  in a basement office:1.3) with (fluorescent lights:1.2). She has (large dark eyes:1.8) and (very short hair:1.2), (brown hair:1.4). She is (wearing glasses:1.1) and a (orange hooded sweatshirt:1.2), with her hands in her pockets. She is (aloof:1.2)
> Dramatic illustration (art by Todd McFarlane:1.6) of ($\color{purple}{\textsf{Amanda Seyfried}}$:1.4) as a programmer (standing  in a basement office:1.3) with (fluorescent lights:1.2). She has (large dark eyes:1.8) and (very short hair:1.2), (brown hair:1.4). She is (wearing glasses:1.1) and a (orange hooded sweatshirt:1.2), with her hands in her pockets. She is (aloof:1.2)
> Dramatic illustration (art by Todd McFarlane:1.6) of ($\color{blue}{\textsf{Cara Delevingne}}$:1.4) as a programmer (standing  in a basement office:1.3) with (fluorescent lights:1.2). She has (large dark eyes:1.8) and (very short hair:1.2), (brown hair:1.4). She is (wearing glasses:1.1) and a (orange hooded sweatshirt:1.2), with her hands in her pockets. She is (aloof:1.2)
> Dramatic illustration (art by Todd McFarlane:1.6) of ($\color{green}{\textsf{Adriana Lima}}$:1.4) as a programmer (standing in a basement office:1.3) with (fluorescent lights:1.2). She has (large dark eyes:1.8) and (very short hair:1.2), (brown hair:1.4). She is (wearing glasses:1.1) and a (orange hooded sweatshirt:1.2), with her hands in her pockets. She is (aloof:1.2)

![2024-03-12_15-36-58_6371](https://github.com/lllyasviel/Fooocus/assets/33701300/d564b91f-f871-46de-8e7d-35cb816d301b)

```
{
"prompt": "Dramatic illustration (art by Todd McFarlane:1.6) of (Anya Taylor-Joy:1.4) as a programmer (standing in a basement office:1.3) with (fluorescent lights:1.2). She has (large dark eyes:1.8) and (very short hair:1.2), (brown hair:1.4). She is (wearing glasses:1.1) and a (orange hooded sweatshirt:1.2), with her hands in her pockets. She is (aloof:1.2)\nDramatic illustration (art by Todd McFarlane:1.6) of (Amanda Seyfried:1.4) as a programmer (standing in a basement office:1.3) with (fluorescent lights:1.2). She has (large dark eyes:1.8) and (very short hair:1.2), (brown hair:1.4). She is (wearing glasses:1.1) and a (orange hooded sweatshirt:1.2), with her hands in her pockets. She is (aloof:1.2)\nDramatic illustration (art by Todd McFarlane:1.6) of (Cara Delevingne:1.4) as a programmer (standing in a basement office:1.3) with (fluorescent lights:1.2). She has (large dark eyes:1.8) and (very short hair:1.2), (brown hair:1.4). She is (wearing glasses:1.1) and a (orange hooded sweatshirt:1.2), with her hands in her pockets. She is (aloof:1.2)\nDramatic illustration (art by Todd McFarlane:1.6) of (Adriana Lima:1.4) as a programmer (standing in a basement office:1.3) with (fluorescent lights:1.2). She has (large dark eyes:1.8) and (very short hair:1.2), (brown hair:1.4). She is (wearing glasses:1.1) and a (orange hooded sweatshirt:1.2), with her hands in her pockets. She is (aloof:1.2)",
"negative_prompt": "logo, patch, unzipped",
"prompt_expansion": "",
"styles": "['SAI Comic Book', 'Flat 2d Art']",
"performance": "Quality",
"resolution": "(1152, 896)",
"guidance_scale": 4,
"sharpness": 2,
"adm_guidance": "(1.5, 0.8, 0.3)",
"base_model": "juggernautXL_v8Rundiffusion.safetensors",
"refiner_model": "None",
"refiner_switch": 0.5,
"sampler": "dpmpp_2m_sde_gpu",
"scheduler": "karras",
"seed": 1824164093032139587,
"lora_combined_1": "EldritchComicsXL1.2.safetensors : 0.6",
"metadata_scheme": false,
"version": "Fooocus v2.2.1"
}
```

**Reply by bblackmoor** [reply]
Is this actually what we are supposed to do?
